web/src/lib/utils/asset-utils.ts

```diff
@@ -103,8 +103,17 @@ export async function bulkDownload(
  * an empty string when not found.
  */
 export function getFilenameExtension(filename: string): string {
-	const lastIndex = filename.lastIndexOf('.');
-	return filename.slice(lastIndex + 1).toLowerCase();
+	const lastIndex = Math.max(0, filename.lastIndexOf('.'));
+	const startIndex = (lastIndex || Infinity) + 1;
+	return filename.slice(startIndex).toLowerCase();
+}
+
+/**
+ * Returns the filename of an asset including file extension
+ */
+export function getAssetFilename(asset: AssetResponseDto): string {
+	const fileExtension = getFilenameExtension(asset.originalPath);
```

Comment:
Should we consider storing the full filename + extension during upload in `asset.originalFileName`? It seems quite annoying to have to constantly grab it from `originalPath`.

Reply:
Clients already send filename + extension, but the server only keeps the filename. Storing both in `originalFileName` makes sense, though changing it might give some trouble on mobile